src/input/return_enums.rs

```diff
@@ -128,7 +128,9 @@ pub(crate) fn validate_iter_to_vec<'py>(
 ) -> ValResult<Vec<PyObject>> {
     let mut output: Vec<PyObject> = Vec::with_capacity(capacity);
     let mut errors: Vec<ValLineError> = Vec::new();
-    for (index, item_result) in iter.enumerate() {
+
+    for (index, is_last_partial, item_result) in state.enumerate_last_partial(iter) {
+        state.allow_partial = is_last_partial && validator.supports_partial();
```

Comment:
Why is the `.supports_partial()` guard necessary (here and in general)?

Reply:
because if you enabled `allow_partial` for a validator which doesn't have proper support for the feature, you would end up skipping errors in the wrong places.

E.g. if you had `list[tuple[list[int], list[str]]`:
* the outer list validator supports `allow_partial`, so it'll ignore errors in the last entry in the input list
* but if it passed `allow_partial=true` down to the tuple validator (which doesn't current support) allow_partial
* it would pass `state` through to both inner list validators unchanged
* then the list validator associated with the 0th entry in the tuple would get `allow_partial=true`
* and therefore errors in the last entry of the 0th member of the tuple in the last entry of the outer list would be ignored when they shouldn't be

Reply:
It feels to me like the "doesn't support partial" is the exception rather than the rule, and probably only for non-variadic tuples? Otherwise, I would have thought that:
- collections would always want to respect `allow_partial` when passed in (forwarding it to their last element)
- everything else e.g. `with_default` etc should not even care about `allow_partial` and just forward it naively